docs/content/why-temporal.md

```diff
@@ -0,0 +1,150 @@
+
+**Scope**
+
+With a traditional system, a service function execution can at best represent a business process.
+Typically, it represents only a part of a business process.
+
+A Temporal Workflow Execution can represent a business process or an entire business object.
+
+### Example subscription use-case
+
+Let's look at a subscription-based use-case to compare the difference between a Temporal Application and other traditional modern approaches.
+
+The basic business steps are as follows:
+
+1. A customer signs up for a service that has a trial period.
+2. After the trial period, if the customer has not cancelled, they should be charged once a month, for up to X months.
+3. The customer has to be notified via email about the charges and should be able to cancel the subscription at any time.
+
+This business logic is not very complicated and can be expressed in a few dozen lines of code.
+In addition to that, any practical implementation has to ensure that the business process is fault-tolerant and scalable.
+
+**Database-centric design approach**
+
+The first approach might be to center everything around a database where an application process would periodically scan the database tables for customers in specific states, execute necessary actions, and update the database to reflect changes.
+
+However, there are various drawbacks.
+
+1. The most obvious one is that the application state machine of the customer's state quickly becomes extremely complicated.
+   For example, if a credit card charge attempt fails or sending an email fails due to a downstream system's unavailability, the state is now in limbo.
+2. Failed calls likely need to be retried for a long time, and these calls need to be throttled to not overload external resources.
+3. There needs to be logic to handle corrupted customer records to avoid blocking the whole process.
+4. Additionally, databases have performance and scalability limitations (eventually requiring sharding) and are not efficient for scenarios that require constant polling.
+
+**Queue system design approach**
+
+The next commonly employed approach is to use a timer service and queues.
+Updates are pushed to a queue while a service consumes them one at a time, updating a database, and possibly pushing more messages into other downstream queues.
+A timer service can be used to schedule queue polling or database actions.
+
+While this approach has shown to scale a bit better, the programming model can become very complex and error-prone, as there are usually no transactional updates between a queuing system, a timer service, and a database.
+
+**Temporal design approach**
+
+The Temporal Platform approach aims to encapsulate and implement the entire business logic in a simple function or object method.
+Thanks to the Temporal Platform, the function/method is stateful, and the implementer doesn't need to employ any additional systems to ensure durability and fault tolerance.
```

Comment:
```suggestion
Thanks to the Temporal Platform, the function/method is durably stateful, and the implementer doesn't need to employ any additional systems to ensure consistency and fault tolerance.
```

My reaction to "method is stateful" is like "of course methods have state/variables". Want to get across that each step/state change is persisted.